src/main/plugins/org.dita.xhtml/xsl/xslhtml/troubleshootingdisplay.xsl

```diff
@@ -0,0 +1,162 @@
+  <xsl:call-template name="generateItemGroupTaskElement"/>
+</xsl:template>
+
+  <xsl:template match="*[contains(@class,' troubleshooting/troubleSolution ')]" name="topic.troubleshooting.troubleSolution">
+  <xsl:call-template name="generateItemGroupTaskElement"/>
+</xsl:template>
+
+<xsl:template name="generateItemGroupTaskElement">
+  <div>
+    <xsl:call-template name="commonattributes"/>
+    <xsl:call-template name="setidaname"/>
+    <xsl:apply-templates/>
+  </div>
+</xsl:template>
+
+  <xsl:template match="*[contains(@class,' troubleshooting/condition ')]" mode="dita2html:section-heading">
+  <xsl:apply-templates select="." mode="generate-task-label">
+    <xsl:with-param name="use-label">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'trouble_condition'"/>
+      </xsl:call-template>
+    </xsl:with-param>
+  </xsl:apply-templates>
+</xsl:template>
+
+  <xsl:template match="*[contains(@class,' troubleshooting/cause ')]" mode="dita2html:section-heading">
+  <xsl:apply-templates select="." mode="generate-task-label">
+    <xsl:with-param name="use-label">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'trouble_cause'"/>
+      </xsl:call-template>
+    </xsl:with-param>
+  </xsl:apply-templates>
+</xsl:template>
+  
+  <xsl:template match="*[contains(@class,' troubleshooting/remedy ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_remedy'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' troubleshooting/responsibleParty ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_responsibleParty'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' task/tasktroubleshooting ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_task'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+
+<!-- 
+     To override the task label for a specific element, match that element with this mode. 
+     For example, you can turn off labels for <context> with this rule:
+     <xsl:template match="*[contains(@class,' task/context ')]" mode="generate-task-label"/>
+-->
+<xsl:template match="*" mode="generate-task-label">
```

Comment:
This should be removed, or else modified so that it uses a new mode name; `generate-task-label` is already part of the task module, so having it here means the original is ignored.

I now see that this adds an extra `troubleshootinglabel` class -- so having this here is currently adding that class to every task heading as well.